Sprint-1/implement/sum.js

```diff
@@ -1,4 +1,9 @@
-function sum(elements) {
-}
-
-module.exports = sum;
+function sum(arr) {
+    if (!Array.isArray(arr)) {
+      throw new TypeError("Input must be an array");
+    }
+    return arr.reduce((total, item) => (typeof item === "number" ? total + item : total), 0);
```

Comment:
What do you expect from the following function calls (on extreme cases)?
Does your function returns the value you expected?

```
sum([Infinity, 1]);
sum([Infinity, Infinity]);
sum([NaN, 1]);
sum([Infinity, -Infinity]);
```

Reply:
sum([Infinity, 1])

Expected: Infinity
Reasoning: Adding any finite number to Infinity results in Infinity.
sum([Infinity, Infinity])

Expected: Infinity
Reasoning: Adding Infinity to Infinity still results in Infinity.
sum([NaN, 1])

Expected: NaN
Reasoning: Adding NaN to any number results in NaN.
sum([Infinity, -Infinity])

Expected: NaN